client_cli/src/main.rs

```diff
@@ -158,19 +168,34 @@ fn main() -> Result<()> {
                 .as_ref(),
         )?
     };
+
     let Configuration(config) = config;
-    println!(
-        "User: {}@{}",
-        config.account_id.name, config.account_id.domain_id
-    );
-    #[cfg(debug_assertions)]
-    eprintln!(
-        "{}",
-        &json5::to_string(&config).wrap_err("Failed to serialize configuration.")?
-    );
-    #[cfg(not(debug_assertions))]
-    eprintln!("This is a release build, debug information omitted from messages");
-    subcommand.run(&config)?;
+
+    if verbose {
+        eprintln!(
+            "User: {}@{}",
+            config.account_id.name, config.account_id.domain_id
+        );
+        #[cfg(debug_assertions)]
+        eprintln!(
+            "{}",
+            &json5::to_string(&config).wrap_err("Failed to serialize configuration.")?
+        );
+        #[cfg(not(debug_assertions))]
+        eprintln!("This is a release build, debug information omitted from messages");
+    }
+
+    let subcommand_output = subcommand.run(&config)?;
+    if json {
+        println!("{}", serde_json::to_string_pretty(&subcommand_output)?)
+    } else {
+        println!(
+            "{}",
+            ron::Options::default()
```

Comment:
The easiest way was to just rely on serde for both machine- and human-facing outputs.
I opted for [RON](https://docs.rs/ron/latest/ron/) for human-readable output, since it resembles what we had earlier (rust debug output) the most. In principle it could be any human-oriented format that's implemented for serde, like [gura](https://gura.netlify.app/)

Reply:
For completeness, here's our other options
- S-expressions would be fast and arguably readable. It's the one format I think could be used for both purposes, but neither as fast as JSON, nor as compact and readable as TOML. 
- JSON is a bit noisy, but it's a de-facto standard. It will look great once we've removed all the redundancies like `Identifiable` from the serialization `impl`s. 
- YAML is terrible format that needs to die. Far too much complexity traded for Python-like syntax. The only advantage those have over curly-braced JSON is that you have to highlight whitespace in order to determine if the nesting is correct. 
- XML is considered harmful. There are good reasons to use it, like for example wanting your data to be extensible, and mostly text. This application has a reputation for being [most unwise](http://harmful.cat-v.org/software/xml/). 
- RON is a 1/1 mapping onto Rust. It's very useful if we want to keep typing information and avoid redundancy. It was the **right call** to prefer this for machine readable interactions, even if this adds to the code size. 
- TOML is useful where data is concentrated on one level, e.g. flat or singly nested structures like our configuration file(s), or is extremely sparse (e.g. 
```toml
[Register.Identifiable.Domain]
id = "wonderland"
metadata = {}
```
Sadly, TOML is not the best option for our purposes, given that the syntax extensions it offers don't help for the kind of data that `iroha_client_cli` will return.

Reply:
I'd also encourage everyone to think about the effects of introducing yet another whitespace-sensitive language into Iroha. We have enough problems as it is with people not being able to decode our error messages. Adding whitespace errors into the mix might not be worth the extra load put on @6r1d

Reply:
Re: whitespace - if we don't accept any input in hypothetical whitespace-sensitive language, we should be fine.
RON looks more natural to me anyways though.

Re: s-exps - as a Vim user I would argue they aren't readable :)

Reply:
RON looks like fine choose, however i don't think that it is all that different from JSON in terms of readability.

Reply:
IMO the fewer formats we use the better. The only exception is the DSL, because it's tailored to our use case. RON is fine because we were already using it. Shall we reconsider and move to JSON @Erigara ?